synapse/rest/media/v1/media_repository.py

```diff
@@ -122,11 +127,32 @@ def __init__(self, hs: "HomeServer"):
             self._start_update_recently_accessed, UPDATE_RECENTLY_ACCESSED_TS
         )
 
+        # Media retention configuration options
+        self._media_retention_local_media_lifetime_ms = (
+            hs.config.media.media_retention_local_media_lifetime_ms
+        )
+        self._media_retention_remote_media_lifetime_ms = (
+            hs.config.media.media_retention_remote_media_lifetime_ms
+        )
+
+        if hs.config.media.media_retention_enabled:
```

Comment:
I suspect we only want this to run on a single worker? Can we put it on the background worker?

Reply:
Is it possible on the background worker? Or must it be on the media worker? Because it need access to the files.

Reply:
Ah, right this touches the actual files. Good call! 👍

Reply:
Yes, this will need to be run on the media worker, as the background worker and media worker may be on different boxes.

That being said, it's a good idea for me to document which worker this task will run on!